### PR TITLE
Enable noImplicitOverride tsc flag.

### PR DESCRIPTION
**Pull request by 12wrigja**
Also sort the keys in tsconfig.json to make it easier to read and diff between codebases.

### CODE REVIEW

**Review by justingrant — APPROVED** [review]
Cool!  The overrides make things clearer.

**Comment by justingrant** [comment]
Also, looks like this needs a rebase for tsconfig.json

**Review by ptomato — APPROVED** [review]

File: tsconfig.json [hunk 1]
```diff
@@ -4,29 +4,29 @@
     "allowUnreachableCode": false,
     "downlevelIteration": true,
     "importHelpers": true,
-    "sourceMap": true,
+    "importsNotUsedAsValues": "error",
     "inlineSources": false,
+    "lib": ["dom", "ES2020"],
```
Comment (justingrant) [inline tsconfig.json:9]: Hey James - Sorry for not catching this earlier. Does this config mean that the polyfill won't support downlevel clients that don't have ES2020?

Comment (12wrigja) [reply tsconfig.json:9]: No - this is just the tsconfig. This output is not published to NPM untouched - it passes through Babel first. We downlevel some of the artifacts published to NPM using Babel: https://github.com/js-temporal/temporal-polyfill/blob/d8dfacb47a562c127156bb53f2790e66d973181d/rollup.config.js#L147

Comment (12wrigja) [reply tsconfig.json:9]: And this particular line was already present in tsconfig (see L19 on the "before" side).

Comment (justingrant) [reply tsconfig.json:9]: OK cool.

**Pull request merged**